src/App.js

```diff
@@ -0,0 +1,408 @@
+  const [advice, setAdvice] = useState("");
+  const [showConfirmation, setShowConfirmation] = useState(false);
+  const [deleteCardId, setDeleteCardId] = useState(null);
+  const [sortOrder, setSortOrder] = useState("desc");
+  const [sortOption, setSortOption] = useState("likes");
+
+  useEffect(() => {
+    fetchCards();
+    fetchBoards();
+  }, []);
+
+  async function fetchCards() {
+    try {
+      const response = await axios.get("https://in-your-face-back-end.onrender.com/cards");
+      setCards(response.data);
+    } catch (error) {
+      console.log("Error fetching cards", error);
+    }
+  }
+
+  async function fetchBoards() {
+    try {
+      const response = await axios.get("https://in-your-face-back-end.onrender.com/boards");
+      setBoards(response.data);
+    } catch (error) {
+      console.log("Error fetching boards", error);
+    }
+  }
+
+  async function getAdvice() {
+    try {
+      const response = await fetch("https://api.adviceslip.com/advice");
+      const data = await response.json();
+      setAdvice(data.slip.advice);
+    } catch (err) {
+      console.log("Error fetching advice", err);
+    }
+  }
+
+  const handleAdviceClick = () => {
+    getAdvice();
+  };
+
+  const addBoard = (newBoard) => {
+    setBoards((prevBoards) => [...prevBoards, newBoard]);
+  };
+
+  // async function updateBoard(boardId, updatedData) {
+  //   try {
+  //     const response = await axios.put(`"https://in-your-face-back-end.onrender.com/boards/${boardId}`, updatedData);//$ for template literals and string interpolation. Not sure.
+
+  //   } catch (error) {We
+  //     console.log("Error updating board", error);
+  //   }
+  // }
+  
+  // async function updateCard(cardId, updatedData) {
+  //   try {
+  //     const response = await axios.put(`"https://in-your-face-back-end.onrender.com/cards/${cardId}`, updatedData);
+      
+  //   } catch (error) {
+  //     console.log("Error updating card", error);
+  //   }
+  // }
+  
+  // async function deleteCard(cardId) {
+  //   try {
+  //     const response = await axios.delete(`"https://in-your-face-back-end.onrender.com/cards/${cardId}`);
+      
+  //   } catch (error) {
+  //     console.log("Error deleting card", error);
+  //   }
+  // }
+  
+  
+
+  const handleFormToggle = () => {
+    setShowForm((show) => !show);
+    setSelectedBoard("");
+  };
+
+  const handleNewBoardToggle = () => {
+    setShowForm(true);
+    setSelectedBoard("");
+  };
+
+  const handleExistingBoardToggle = (boardName) => {
+    setSelectedBoard(boardName);
+  };
+
+  const handleConfirmDelete = (cardId) => {
+    setDeleteCardId(cardId);
+    setShowConfirmation(true);
+  };
+
+  const handleDeleteConfirmation = (confirmed) => {
+    if (confirmed) {
+      setCards((prevCards) => prevCards.filter((card) => card.id !== deleteCardId));
+    }
+    setShowConfirmation(false);
+  };
+
+  const handleIncrementLikes = (cardId, category) => {
```

Comment:
👀 This is only increments the like count in the local cards state. This doesn't affect any cards in the database.

Consider adding a custom route to the backend for liking a card that only needs the liked card id. Then the UI wouldn't need to know about any of the logic required to like a card. It could call the like endpoint, get the updated card data, and use that to update the local card state.